tests/index-test.js

```diff
@@ -0,0 +1,194 @@
+    expect(() => parseIdentifier('package/./foo')).to.throw(/invalid identifier/);
+    expect(() => parseIdentifier('package/../foo')).to.throw(/invalid identifier/);
+    expect(() => parseIdentifier('@scoped/./bar')).to.throw(/invalid identifier/);
+    expect(() => parseIdentifier('@scoped/../bar')).to.throw(/invalid identifier/);
+
+    expect(parseIdentifier('../foo/_bar')).to.eql({
+      package: null,
+      path: '../foo/_bar',
+    });
+    expect(parseIdentifier('./foo/_bar')).to.eql({
+      package: null,
+      path: './foo/_bar',
+    });
+
+    expect(parseIdentifier('@a/a')).to.eql({
+      package: '@a/a',
+      path: null,
+    });
+    expect(parseIdentifier('@a/a/a')).to.eql({
+      package: '@a/a',
+      path: 'a',
+    });
+    expect(parseIdentifier('rsvp')).to.eql({
+      package: 'rsvp',
+      path: null,
+    });
+
+    expect(parseIdentifier('@rsvp/all')).to.eql({
+      package: '@rsvp/all',
+      path: null,
+    });
+
+    expect(parseIdentifier('rsvp/index')).to.eql({
+      package: 'rsvp',
+      path: 'index',
+    });
+
+    expect(parseIdentifier('rsvp/utils/index')).to.eql({
+      package: 'rsvp',
+      path: 'utils/index',
+    });
+
+    expect(parseIdentifier('@rsvp/all/index')).to.eql({
+      package: '@rsvp/all',
+      path: 'index',
+    });
+    expect(parseIdentifier('@rsvp/all/utils/index')).to.eql({
+      package: '@rsvp/all',
+      path: 'utils/index',
+    });
+  });
+
+  it('throws an exception if the identifier is malformed', function () {
+    expect(() => resolve('my addon/_person.graphql', { basedir: project.baseDir })).to.throw(
+      /invalid identifier: 'my addon\/_person.graphql'/,
+    );
+  });
+
+  it('supports basic v1 addon resolving', function () {
+    expect(
+      resolve('my-addon/_person.graphql', {
+        basedir: project.baseDir,
+      }),
+    ).to.eql(`${myAddon.baseDir}/addon/_person.graphql`);
+  });
+
+  it('supports basic v2 addon resolving', function () {
+    expect(
+      resolve('my-v2-addon/_person.graphql', {
+        basedir: project.baseDir,
+      }),
+    ).to.eql(`${myV2Addon.baseDir}/_person.graphql`);
+  });
+
+  it('returns null if no such package exists but the syntax is valid', function () {
+    expect(
+      resolve('not-a-real-addon/_person.graphql', {
+        basedir: project.baseDir,
+      }),
+    ).to.eql(null);
+  });
+
+  it('resolves in-repo v1 addons from the root', function () {
+    expect(
+      resolve('the-in-repo-addon/_person.graphql', {
+        basedir: project.baseDir,
+      }),
+    ).to.eql(path.join(project.baseDir, '/in-repo-addons/the-in-repo-addon/addon/_person.graphql'));
+  });
+
+  it('resolves in-repo v1 addons from a nested path', function () {
+    expect(
+      resolve('the-in-repo-addon/_person.graphql', {
+        // TODO: maybe replace this with a filename e.g. ${project}/app/routes/my-query.graphql
+        basedir: `${project.baseDir}/app/routes`,
```

Comment:
@stefanpenner this test would fail in master because `current = resolvePackagePath(basedir)` would fail to find the project's `package.json`